libs/datamodel/connectors/sql-datamodel-connector/src/mysql_datamodel_connector.rs

```diff
@@ -156,45 +156,64 @@ impl Connector for MySqlDatamodelConnector {
     }
 
     fn validate_field(&self, field: &Field) -> Result<(), ConnectorError> {
-        if let FieldType::NativeType(_, native_type) = field.field_type() {
-            let native_type_name = native_type.name.as_str();
-            if matches!(native_type_name, DECIMAL_TYPE_NAME | NUMERIC_TYPE_NAME) {
-                match native_type.args.as_slice() {
```

Comment:
This match encodes how many args we expect. This is a bit ugly and is solved by this PR. Right now it is not a big problem yet but becomes one once args can be more than just integers.